lib/Conversion/TTIRToTTMetal/TTIRToTTMetal.cpp

```diff
@@ -741,36 +750,137 @@ class TTIRToTTMetalDispatchRewriter : public OpRewritePattern<ttir::GenericOp> {
+
+      SmallVector<std::int64_t> operandIndicesRecip;
+      // For DIV, input 1 is going through reciprocal.
+      operandIndicesRecip.push_back(1);
+      commonComputeRecipOp(arithOrMathOp, cbOperands, iterators,
+                           blockArgIteratorMapping, builder,
+                           operandIndicesRecip);
+
+      builder.create<ttkernel::MulTilesInitOp>(arithOrMathOp.getLoc(), inCB0,
+                                               inCB1);
+
+      commonComputeMulOp(arithOrMathOp, cbOperands, iterators,
+                         blockArgIteratorMapping, builder);
+
+      Value one = i32(1, builder);
+      builder.create<ttkernel::CBPopFrontOp>(arithOrMathOp.getLoc(), inCB1,
+                                             one);
     } else {
-      llvm_unreachable("Unhandled binary op compute conversion.");
+      llvm_unreachable("Unhandled conversion for operation which is neither "
+                       "unary nor binary.");
     }
   }
 
+  void commonComputeMulOp(Operation &op, ArrayRef<BlockArgument> cbOperands,
+                          ArrayRef<BlockArgument> iterators,
+                          SmallVector<unsigned> blockArgIteratorMapping,
+                          OpBuilder &builder) const {
+
+    auto inCB0 = cbOperands[0];
+    auto inCB1 = cbOperands[1];
+    auto outCB = cbOperands[2];
+
+    Value dstIndex = i32(0, builder);
+
+    builder.create<ttkernel::TileRegsAcquireOp>(op.getLoc());
+    if (mlir::isa<arith::MulFOp>(op)) {
+      builder.create<ttkernel::MulTilesOp>(
+          op.getLoc(), inCB0, inCB1, iterators[blockArgIteratorMapping[0]],
+          iterators[blockArgIteratorMapping[0]], dstIndex);
+    } else if (mlir::isa<arith::DivFOp>(op)) {
+      builder.create<ttkernel::MulTilesOp>(
+          op.getLoc(), inCB0, inCB1, iterators[blockArgIteratorMapping[0]],
+          dstIndex, dstIndex);
+    } else {
+      llvm_unreachable("Common compute for multiplying tiles should be called "
+                       "only on MulFOp and DivFOp");
+    }
+
+    builder.create<ttkernel::TileRegsCommitOp>(op.getLoc());
+    builder.create<ttkernel::TileRegsWaitOp>(op.getLoc());
+    builder.create<ttkernel::PackTileOp>(op.getLoc(), dstIndex, outCB,
+                                         iterators[blockArgIteratorMapping[2]]);
+    builder.create<ttkernel::TileRegsReleaseOp>(op.getLoc());
+  }
+
+  void commonComputeRecipOp(Operation &op, ArrayRef<BlockArgument> cbOperands,
+                            ArrayRef<BlockArgument> iterators,
+                            SmallVector<unsigned> blockArgIteratorMapping,
+                            OpBuilder &builder,
+                            SmallVector<std::int64_t> &operandIndices) const {
+    Value dstIndex = i32(0, builder);
+    Value one = i32(1, builder);
+
+    builder.create<ttkernel::CopyTileInitOp>(op.getLoc());
+    builder.create<ttkernel::CBReserveBackOp>(
+        op.getLoc(), cbOperands[operandIndices[0]], one);
```

Comment:
IMO it would be much more obvious what is going on just by using `inCB1` instead of `cbOperands[operandIndices[0]]`.

Reply:
I have added inputCB and outputCB, it is not always going to be inCB1, you can sometimes use recip on some other CB if you want

Reply:
Yes but in this concrete op, which will have its own converter when we do #807, inCB1 will be used always, right?
I'd much rather be explicit than save a couple of almost-duplicated lines.
But lets discuss that when #807  starts.